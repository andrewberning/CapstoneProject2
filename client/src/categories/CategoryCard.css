.CategoryCard {
  width: 250px;
  height: 400px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 10px;
}

.CategoryCard a {
  text-decoration: none;
}

.CategoryCard a:hover {
  text-decoration: underline;
}

.category-img-container {
  flex-grow: 1;
  display: flex;
  align-items: center;
  justify-content: center;
}

.card-img {
  max-width: 100%;
  max-height: 100%;
  object-fit: contain;
}

.card-body {
  flex-grow: 0;
  padding: 10px;
}