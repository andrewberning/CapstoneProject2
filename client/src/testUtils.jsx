import React from "react";
import UserContext from "./auth/UserContext";
import { vi } from 'vitest';

const demoUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "testuser123@test.net",
};

// Mock functions
const mockSetCurrentUser = vi.fn();
const mockLogin = vi.fn();
const mockLogout = vi.fn();
const mockSignup = vi.fn();
const mockAddToCart = vi.fn();
const mockHasAddedToCart = vi.fn();

const UserProvider = ({ 
  children,
  currentUser = demoUser,
  setCurrentUser = mockSetCurrentUser,
  login = mockLogin,
  logout = mockLogout,
  signup = mockSignup,
  addToCart = mockAddToCart,
  hasAddedToCart = mockHasAddedToCart,
  cartItems = [],
  setCartItems = () => {},
  categories = [],
  infoLoaded = true,
  totalAmount = 0,
  totalItems = 0,
}) => (
  <UserContext.Provider value={{ 
    currentUser,
    setCurrentUser,
    login,
    logout,
    signup,
    addToCart,
    hasAddedToCart,
    cartItems,
    setCartItems,
    categories,
    infoLoaded,
    totalAmount,
    totalItems,
  }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider, mockSetCurrentUser, mockLogin, mockLogout, mockSignup, mockAddToCart, mockHasAddedToCart };
