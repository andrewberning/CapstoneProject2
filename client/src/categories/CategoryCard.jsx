import { Link } from "react-router-dom";
import "./CategoryCard.css";

export default function CategoryCard({ category, image }) {
  return (
    <li className="CategoryCard">
      <Link className="category-link card" to={`/categories/${category}`}>
        <div className="category-img-container">
          <img src={image} alt={category} className="card-img" />
        </div>
        <div className="card-body">
          <div className="card-title text-center">{category}</div>
        </div>
      </Link>
    </li>
  );
}