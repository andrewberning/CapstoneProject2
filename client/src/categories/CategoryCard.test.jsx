import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CategoryCard from "./CategoryCard";
import { UserProvider } from "../testUtils";

describe('CategoryCard component', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider >
          <CategoryCard />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <CategoryCard />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});