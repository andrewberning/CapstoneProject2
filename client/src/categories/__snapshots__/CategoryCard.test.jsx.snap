// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoryCard component > matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="CategoryCard"
  >
    <a
      class="category-link card flex-column"
      href="/categories/undefined"
    >
      <div
        class="category-img-container"
      >
        <img
          class="card-img"
        />
      </div>
      <div
        class="card-body"
      >
        <div
          class="card-title text-center"
        />
      </div>
    </a>
  </li>
</DocumentFragment>
`;
